### PR TITLE
add optimize-css-assets-webpack-plugin

### PR DESCRIPTION
**Pull request by amirfefer**
Supported Versions:

 * Nightly

### CODE REVIEW

**Comment by tbrisker** [comment]
@amirfefer you need to bump the release and add a changelog entry

**Comment by tbrisker** [comment]
looks like an actual build error now related to the change:
```
Error: Cannot find module 'cssnano'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at module.exports.env (/builddir/build/BUILD/foreman-2.6.0-develop/config/webpack.config.js:216:23)
    at handleFunction (/opt/theforeman/tfm/root/usr/lib/node_modules/webpack/lib/prepareOptions.js:26:13)
    at prepareOptions (/opt/theforeman/tfm/root/usr/lib/node_modules/webpack/lib/prepareOptions.js:11:13)
    at requireConfig (/opt/theforeman/tfm/root/usr/lib/node_modules/webpack/bin/convert-argv.js:98:14)
    at /opt/theforeman/tfm/root/usr/lib/node_modules/webpack/bin/convert-argv.js:104:17
    at Array.forEach (<anonymous>)
    at module.exports (/opt/theforeman/tfm/root/usr/lib/node_modules/webpack/bin/convert-argv.js:102:15)
    at yargs.parse (/opt/theforeman/tfm/root/usr/lib/node_modules/webpack/bin/webpack.js:171:41)
    at Object.Yargs.self.parse (/opt/theforeman/tfm/root/usr/lib/node_modules/webpack/node_modules/yargs/yargs.js:533:18)
    at Object.<anonymous> (/opt/theforeman/tfm/root/usr/lib/node_modules/webpack/bin/webpack.js:152:7)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)

**Comment by tbrisker** [comment]
@amirfefer please update again to include cssnano

File: packages/foreman/foreman/foreman.spec [hunk 1]
```diff
@@ -270,6 +270,8 @@ BuildRequires: %{?scl_prefix}npm(compression-webpack-plugin) >= 1.1.11
 BuildRequires: %{?scl_prefix}npm(compression-webpack-plugin) < 1.2.0
 BuildRequires: %{?scl_prefix}npm(css-loader) >= 0.23.1
 BuildRequires: %{?scl_prefix}npm(css-loader) < 1.0.0
+BuildRequires: %{?scl_prefix}npm(cssnano) >= 3.4.0
+BuildRequires: %{?scl_prefix}npm(cssnano) < 4.1.1
```
Comment (tbrisker) [inline packages/foreman/foreman/foreman.spec:274]: This doesn't match the version in core: https://github.com/theforeman/foreman/blob/develop/package.json#L43
Also, it's missing from the `Requires` section below

File: packages/foreman/foreman/foreman.spec [hunk 2]
```diff
@@ -1024,6 +1030,10 @@ exit 0
 %systemd_postun %{name}.socket
 
 %changelog
+* Wed Jun 02 2021 Amir Feferkuchen <afeferku@redhat.com> - 2.6.0-0.2.develop
+- Add optimize-css-assets-webpack-plugin
+- Update depdendencies cssnano
```
Comment (tbrisker) [inline packages/foreman/foreman/foreman.spec:1035]: How about:
```suggestion
- Update NPM dependencies
```

**Review by tbrisker — CHANGES_REQUESTED** [review]
Please update the commit message to match the current content

**Comment by tbrisker** [comment]
@amirfefer the other comments still need fixing, and the versions still don't match what is in the package.json

**Comment by amirfefer** [comment]
@tbrisker, thanks 👍 
updated

**Review by tbrisker — APPROVED** [review]

**Pull request merged**